tests/backup_helper.go

```diff
@@ -10471,3 +10493,341 @@ func IsPxInstalled() bool {
+		},
+		// userid of the user(s) to share the cluster with
+		Users: users,
+		// group(s) to share the cluster with
+		Groups: groups,
+	}
+	resp, err := backupDriver.UnShareCluster(ctx, unshareClusterRequest)
+	if err != nil {
+		return resp, err
+	}
+
+	err = ValidateUnShareCluster(ctx, clusterName, clusterUid, users, groups)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// ValidateShareCluster validates that a cluster is shared with users and groups by validating collaborators list and inspecting cluster from user.
+func ValidateShareCluster(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string) error {
+	backupDriver := Inst().Backup
+	userIds := []string{}
+	groupIds := []string{}
+	req := &api.ClusterInspectRequest{
+		Name:  clusterName,
+		OrgId: BackupOrgID,
+		Uid:   clusterUid,
+	}
+
+	// Inspect cluster to retrieve collaborators
+	clusterObject, err := backupDriver.InspectCluster(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	// Extract collaborator user and group IDs
+	clusterCollaborators := clusterObject.GetCluster().Ownership
+	for _, user := range clusterCollaborators.Collaborators {
+		userIds = append(userIds, user.Id)
+	}
+	for _, group := range clusterCollaborators.Groups {
+		groupIds = append(groupIds, group.Id)
+	}
+
+	// Validate that users and groups are present in the collaborators list
+	for _, user := range users {
+		if !IsPresent(userIds, user) {
+			return fmt.Errorf("User with id [%s] is not present in the list of collaborators for cluster [%s]", user, clusterName)
+		}
+	}
+	for _, group := range groups {
+		if !IsPresent(groupIds, group) {
+			return fmt.Errorf("Group with id [%s] is not present in the list of collaborators for cluster [%s]", group, clusterName)
+		}
+	}
+
+	// Validate that the entire list matches
+	if !AreStringSlicesEqual(userIds, users) {
+		return fmt.Errorf("User list [%v] does not match the collaborators users list [%v] for cluster [%s]", users, userIds, clusterName)
+	}
+
+	if !AreStringSlicesEqual(groupIds, groups) {
+		return fmt.Errorf("Group list [%v] does not match the collaborators group list [%v] for cluster [%s]", groups, groupIds, clusterName)
+	}
+
+	// Expand the users list with members from groups
+	for _, group := range groups {
+		usersFromGroup, _ := backup.GetMembersOfGroupWithID(group)
+		for _, user := range usersFromGroup {
+			if !IsPresent(users, user) {
+				users = append(users, user)
+			}
+		}
+	}
+
+	// Validation that the cluster is accessible to shared users
+	errorChan := make(chan error, len(users))
+	defer close(errorChan)
+
+	var wg sync.WaitGroup
+	for _, user := range users {
+		wg.Add(1)
+		go func(user string) {
+			defer wg.Done()
+			nonAdminCtx, err := backup.GetNonAdminCtx(user, CommonPassword)
+			if err != nil {
+				errorChan <- err
+				return
+			}
+			log.Infof("Inspecting cluster [%s] with user [%s]", clusterName, user)
+			_, err = backupDriver.InspectCluster(nonAdminCtx, req)
```

Comment:
I would prefer createBackup, createBackupSchedule and Create restore on this cluster is better validation .

Reply:
All backup-related CRUD operations will be tested at the test case level